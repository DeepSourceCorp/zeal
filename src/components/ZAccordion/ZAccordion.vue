<template>
  <div>
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'ZAccordion',
  components: {},
  props: {
    defaultStyle: {
      type: Boolean,
      default: false
    },
    disabled: {
      type: Boolean,
      default: false
    },
    allowMultiple: {
      type: Boolean,
      default: true
    }
  },
  data() {
    return {
      accordionItems: []
    }
  },
  computed: {}
}
</script>
