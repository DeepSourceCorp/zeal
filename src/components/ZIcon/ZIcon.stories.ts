import ZIcon from './ZIcon.vue'

import '../../assets/css/tailwind.css'
import '../../assets/css/typography.css'
import '../../assets/css/layout.css'

export default {
  title: 'Icons',
  component: ZIcon,
  excludeStories: /.*Data$/
}

export const Default = () => ({
  components: { ZIcon },
  template: `<div class='wrapper'>
        <z-icon icon='arrow-up'></z-icon>
        <p class="icon-title">arrow-up</p>
    </div>`
})

export const IconWithCustomColor = () => ({
  components: { ZIcon },
  template: `<div class="wrapper">
        <z-icon icon="activity" color="juniper"></z-icon>
        <p class="icon-title">activity</p>
    </div>`
})

export const SmallIcon = () => ({
  components: { ZIcon },
  template: `<div class='wrapper'>
        <z-icon icon='arrow-up' size="small"></z-icon>
        <p class='icon-title'>arrow-up</p>
    </div>`
})
