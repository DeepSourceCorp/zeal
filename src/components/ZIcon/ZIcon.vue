<template>
    <svg 
      :class="[
        'z-icon',
        `${size && `z-icon--${size}`}`,
        `${icon && `z-icon--${icon}`}`
      ]"
      viewBox="0 0 24 24" 
      v-html="getIcon(icon)"></svg>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'
import feather from 'feather-icons'
import customIcons from '../../utils/icon-helper'

@Component({
  name: 'ZIcon'
})
export default class Icon extends Vue {
  @Prop({ default: '' })
  private icon!: string
  @Prop({default: ''})
  private size!: string

  public getIcon(iconName: string): HTMLOrSVGImageElement {
    let DOM = null
    if (feather.icons[iconName]) {
      DOM = feather.icons[iconName] && feather.icons[iconName].contents
    } else {
      DOM = customIcons[iconName] &&  customIcons[iconName].contents;
    }
    return DOM
  }
}
</script>

<style lang="css" scoped>
  .z-icon {
    @apply  w-6 h-6 fill-current stroke-current text-slate stroke-2;
  }

  .z-icon--small {
    @apply w-4 h-4;
  }
</style>
