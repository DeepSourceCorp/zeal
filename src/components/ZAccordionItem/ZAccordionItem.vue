<template>
  <div
    class="z-accordion-item"
    :class="{
      'border-ink-200 border-t border-b first:border-0 last:border-0 hover:bg-ink-300': this
        .showBorders,
      'text-vanilla-200': !this.isDisabled,
      'text-slate': this.isDisabled
    }"
  >
    <slot name="title" :open="open" :toggleAccordion="toggleAccordion">
      <div
        class="flex items-center transition-all duration-700 ease-in-out group"
        @click="toggleAccordion()"
        :class="{
          'cursor-not-allowed': this.isDisabled,
          'cursor-pointer': !this.isDisabled && !this.isList
        }"
      >
        <span class="flex-1 font-medium">{{ title }}</span>
        <z-icon
          icon="chevron-right"
          color="slate"
          class="transform transition-all duration-DEFAULT ease-in-out group"
          :class="[
            accordionHeaderAnimations,
            {
              'text-vanilla-200 rotate-90': this.open,
              'text-slate': this.isDisabled,
              'group-hover:text-vanilla-200': !this.isDisabled,
              hidden: this.isList
            }
          ]"
        ></z-icon>
      </div>
    </slot>
    <div
      class="overflow-scroll transition-max-height duration-300 ease-in-out text-sm leading-6"
      :class="{
        'max-h-52': this.open,
        'max-h-0': !this.open,
        'max-h-full': this.isList
      }"
    >
      <slot></slot>
    </div>
  </div>
</template>

<script>
import ZIcon from '@/components/ZIcon'
export default {
  name: 'ZAccordionItem',
  components: {
    ZIcon
  },
  props: {
    title: {
      required: true,
      type: String
    },
    isOpen: {
      type: Boolean,
      default: false
    },
    icon: {
      type: String,
      default: 'down'
    },
    isList: {
      type: Boolean,
      default: false
    }
  },
  data() {
    return {
      open: this.isOpen,
      accordionHeaderAnimations: '',
      isDisabled: this.$parent.disabled,
      showBorders: this.$parent.showBorders
    }
  },
  methods: {
    toggleAccordion() {
      if (this.isDisabled) return
      if (this.isList) return
      this.open = !this.open
      this.accordionHeaderAnimations = this.open
        ? 'animate-first-quarter-spin rotate-90'
        : 'animate-reverse-quarter-spin rotate-0'
      this.$emit('isOpen', this.open)
    }
  }
}
</script>
