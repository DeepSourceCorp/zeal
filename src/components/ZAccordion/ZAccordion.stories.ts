import '../../assets/css/tailwind.css'
import '../../assets/css/typography.css'
import '../../assets/css/layout.css'
import ZAccordion from './ZAccordion.vue'
import ZAccordionItem from '@/components/ZAccordionItem/ZAccordionItem.vue'
import ZIcon from '@/components/ZIcon/ZIcon.vue'
import ZButton from '@/components/ZButton/ZButton.vue'
import ZInput from '@/components/ZInput/ZInput.vue'

export default {
  title: 'Accordion',
  component: ZAccordion,
  excludeStories: /.*Data$/
}

export const DefaultAccordion = () => ({
  components: { ZAccordion, ZAccordionItem },
  template: `<div class='padded-container input-container'>
        <z-accordion class="text-vanilla-100">
          <z-accordion-item title="Consistency" class="p-4">
            <div>Consistent with real life: in line with the process and logic of real life, and comply with languages and habits that the users are used to;</div>
            <div>Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc.</div>
          </z-accordion-item>
          <z-accordion-item title="Feedback" class="p-4">
            <div>Consistent with real life: in line with the process and logic of real life, and comply with languages and habits that the users are used to;</div>
            <div>Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc.</div>
          </z-accordion-item>
          <z-accordion-item title="Efficiency" class="p-4">
            <div>Consistent with real life: in line with the process and logic of real life, and comply with languages and habits that the users are used to;</div>
            <div>Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc.</div>
          </z-accordion-item>
        </z-accordion>
    </div>`
})

export const AccordionWithBaseStyle = () => ({
  components: { ZAccordion, ZAccordionItem },
  template: `<div class='padded-container input-container'>
        <z-accordion class="text-vanilla-100" :defaultStyle="true">
          <z-accordion-item title="Consistency" class="p-4">
            <div>Consistent with real life: in line with the process and logic of real life, and comply with languages and habits that the users are used to;</div>
            <div>Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc.</div>
          </z-accordion-item>
          <z-accordion-item title="Feedback" class="p-4">
            <div>Consistent with real life: in line with the process and logic of real life, and comply with languages and habits that the users are used to;</div>
            <div>Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc.</div>
          </z-accordion-item>
          <z-accordion-item title="Efficiency" class="p-4">
            <div>Consistent with real life: in line with the process and logic of real life, and comply with languages and habits that the users are used to;</div>
            <div>Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc.</div>
          </z-accordion-item>
        </z-accordion>
    </div>`
})

export const AccordionWithOpenedCards = () => ({
  components: { ZAccordion, ZAccordionItem },
  template: `<div class='padded-container input-container'>
        <z-accordion class="text-vanilla-100" :defaultStyle="true">
          <z-accordion-item title="Consistency" :is-open="true" class="p-4">
            <div>Consistent with real life: in line with the process and logic of real life, and comply with languages and habits that the users are used to;</div>
            <div>Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc.</div>
          </z-accordion-item>
          <z-accordion-item title="Feedback" class="p-4">
            <div>Consistent with real life: in line with the process and logic of real life, and comply with languages and habits that the users are used to;</div>
            <div>Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc.</div>
          </z-accordion-item>
          <z-accordion-item title="Efficiency" class="p-4">
            <div>Consistent with real life: in line with the process and logic of real life, and comply with languages and habits that the users are used to;</div>
            <div>Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc.</div>
          </z-accordion-item>
        </z-accordion>
    </div>`
})

export const AccordionWithDisabledState = () => ({
  components: { ZAccordion, ZAccordionItem },
  template: `<div class='padded-container input-container'>
        <z-accordion class="text-vanilla-100" :defaultStyle="true" :disabled="true">
          <z-accordion-item title="Consistency" class="p-4">
            <div>Consistent with real life: in line with the process and logic of real life, and comply with languages and habits that the users are used to;</div>
            <div>Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc.</div>
          </z-accordion-item>
          <z-accordion-item title="Feedback" class="p-4">
            <div>Consistent with real life: in line with the process and logic of real life, and comply with languages and habits that the users are used to;</div>
            <div>Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc.</div>
          </z-accordion-item>
          <z-accordion-item title="Efficiency" class="p-4">
            <div>Consistent with real life: in line with the process and logic of real life, and comply with languages and habits that the users are used to;</div>
            <div>Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc.</div>
          </z-accordion-item>
        </z-accordion>
    </div>`
})

export const AccordionUsedInNavBar = () => ({
  components: { ZAccordion, ZAccordionItem },
  data() {
    return {
      links: ['Documentation', 'Discourse Form', 'Learning Center', 'Blog', 'Slack User Group']
    }
  },
  template: `<div class='padded-container input-container'>
        <z-accordion class="text-vanilla-100">
          <z-accordion-item title="Resources" class="font-medium p-4">
            <div class="flex flex-col text-sm space-y-4 py-3">
              <a v-for="link in links" :key="link" href="#" class="hover:text-juniper transition-all duration-75 ease-in-out">{{link}}</a>
            </div>
          </z-accordion-item>
        </z-accordion>
    </div>`
})

export const AccordionUsedInHeader = () => ({
  components: { ZAccordion, ZAccordionItem, ZIcon },
  data() {
    return {
      openAccordion: false,
      links: ['Documentation', 'Discourse Form', 'Learning Center', 'Blog', 'Slack User Group']
    }
  },
  template: `<div class='padded-container flex items-center w-52'>
        <z-accordion class="text-vanilla-100 w-full">
          <z-accordion-item :is-open="openAccordion" class="p-4">
            <template v-slot:title="{ open, toggleAccordion }">
              <div class="flex items-center cursor-pointer w-full" @click="toggleAccordion()">
                <span class="flex-1 font-medium text-slate">Resources</span>
                <z-icon icon="chevron-right" 
                    size="small" color="slate" 
                    class="transform"
                    :class="open ? 'animate-first-quarter-spin rotate-90' : 'animate-reverse-quarter-spin rotate-0'"></z-icon>
              </div>
            </template>
            <div class="flex flex-col text-sm space-y-4 py-3">
              <a v-for="link in links" :key="link" href="#" class="hover:text-juniper transition-all duration-75 ease-in-out">{{link}}</a>
            </div>
          </z-accordion-item>
        </z-accordion>
    </div>`
})

export const ListLikeAccordion = () => ({
  components: { ZAccordion, ZAccordionItem, ZIcon },
  data() {
    return {
      openAccordion: false,
      links: ['Documentation', 'Discourse Form', 'Learning Center', 'Blog', 'Slack User Group']
    }
  },
  template: `<div class='padded-container flex items-center w-52'>
        <z-accordion class="text-vanilla-100 w-full">
          <z-accordion-item :is-open="openAccordion" :is-list="true" title="List Items" class="p-4">
            <div class="flex flex-col text-sm space-y-4 py-3">
              <a v-for="link in links" :key="link" href="#" class="hover:text-juniper transition-all duration-75 ease-in-out">{{link}}</a>
            </div>
          </z-accordion-item>
        </z-accordion>
    </div>`
})

export const WebNextFooter = () => ({
  components: { ZAccordion, ZAccordionItem, ZIcon, ZButton, ZInput },
  data() {
    return {
      openAccordion: false,
      name: '',
      product: ['Autofix', 'Code formatters', 'Pricing', 'Security'],
      languages: ['For Python', 'For Go', 'For Ruby', 'For JavaScript'],
      resources: ['Documentation', 'Blog', 'Changelog', 'Slack user group'],
      company: [
        'About',
        'Customers',
        'Jobs',
        'Privacy Policy',
        'Terms of Service',
        'Press Enquiries',
        'Brand Assets'
      ]
    }
  },
  computed: {
    isMobile() {
      return window.innerWidth < 640
    }
  },
  template: `<div class="h-screen w-full-screen">
        <div class="prose prose-md max-w-none p-20">
        <h1 v-for="i in 8">Some random heading to grab attention</h1>
        </div>
        <footer class="w-full p-12 flex flex-col sm:space-y-0 space-y-2 sm:grid grid-rows-footer grid-cols-footer gap-x-8 gap-y-4 bg-transparent text-vanilla-100 border-t border-ink-200 min-h-102 bg-gradient-dark_dawn animate-gradient bg-400%">
            <div class="flex row-start-1 row-end-1 col-start-1 col-end-1">
                <img class="max-w-none h-8" src="https://i.imgur.com/zKLLWIr.png"/>
            </div>

            <z-accordion-item :is-list="!isMobile" title="Product" class="row-start-1 row-end-1 col-start-2 col-end-2 flex flex-col px-0 py-0 space-y-2 sm:space-y-6 text-vanilla-400 tracking-wide uppercase">
              <div class="flex flex-col space-y-2">
                <a v-for="item in product" :key="item" href="#" class="hover:text-juniper transition-all duration-75 ease-in-out flex flex-col space-y-4 text-vanilla-300 capitalize">
                    {{item}}
                </a>
              </div>
            </z-accordion-item>

            <z-accordion-item :is-list="!isMobile" title="languages" class="row-start-1 row-end-1 col-start-3 col-end-3 flex flex-col px-0 py-0  space-y-2 sm:space-y-6 text-vanilla-400 tracking-wide uppercase">
              <div class="flex flex-col space-y-2">
                <a v-for="language in languages" :key="language" href="#" class="hover:text-juniper transition-all duration-75 ease-in-out flex flex-col space-y-4 text-vanilla-300 capitalize">
                    {{language}}
                </a>
              </div>
            </z-accordion-item>

            <z-accordion-item :is-list="!isMobile" title="resources" class="row-start-1 row-end-1 col-start-4 col-end-4 flex flex-col px-0 py-0  space-y-2 sm:space-y-6 text-vanilla-400 tracking-wide uppercase">
              <div class="flex flex-col space-y-2">
                <a v-for="item in resources" :key="item" href="#" class="hover:text-juniper transition-all duration-75 ease-in-out flex flex-col space-y-4 text-vanilla-300 capitalize">
                    {{item}}
                </a>
              </div>
            </z-accordion-item>

            <z-accordion-item :is-list="!isMobile" title="company" class="row-start-1 row-end-1 col-start-5 col-end-5 flex flex-col px-0 py-0  space-y-2 sm:space-y-6 text-vanilla-400 tracking-wide uppercase">
              <div class="flex flex-col space-y-2">
                <a v-for="item in company" :key="item" href="#" class="hover:text-juniper transition-all duration-75 ease-in-out flex flex-col space-y-4 text-vanilla-300 capitalize">
                    {{item}}
                </a>
              </div>
            </z-accordion-item>

            <div class="row-start-1 row-end-1 col-start-6 col-end-6 flex flex-col space-y-6 text-vanilla-400 text-sm">
                <div class="font-medium uppercase tracking-wide hidden sm:block">Social Media</div>
                <div class="flex flex-col-reverse sm:flex-col gap-y-3 sm:gap-y-16">
                  <div class="flex flex-col space-y-4 items-center sm:items-start">
                    <div class="text-vanilla-300 hidden sm:block">Follow us on social media to stay updated.</div>
                    <div class="flex space-x-3">
                        <z-icon size="medium" icon="dribbble" color="vanilla-300"></z-icon>
                        <z-icon size="medium" icon="dribbble" color="vanilla-300"></z-icon>
                        <z-icon size="medium" icon="dribbble" color="vanilla-300"></z-icon>
                    </div>
                  </div>
                  <div class="flex flex-col space-y-2">
                    <div class="uppercase">Sign up for the newsletter</div>
                    <z-input v-model="name" backgroundColor="transparent">
                      <template slot="right">
                        <z-button color="primary" icon="chevron-right" iconSize="medium" iconColor="ink-400"></z-button>
                      </template>
                    </z-input>
                  </div>
                </div>
            </div>

            <div class="block sm:hidden w-full h-px bg-slate"></div>

            <div class="row-start-1 row-end-1 col-start-7 col-end-7 text-sm text-vanilla-400">
                <div class="flex justify-center flex-row space-x-2 sm:space-x-0 sm:flex-col space-y-0 sm:space-y-2">
                    <div v-for="i in 2" :key="i" class="flex flex-col sm:grid grid-rows-3 grid-cols-3 gap-x-4 p-2 sm:p-2.5 tracking-wide bg-ink-300 rounded-sm w-auto sm:w-full h-auto sm:h-20 flex items-center">
                        <img class="max-w-none row-span-3" src="https://i.imgur.com/JBKMaq5.png"/>
                        <span class="uppercase font-medium col-span-2 text-vanilla-100">GDPR</span>
                        <span class="row-span-2 col-span-2 tracking-normal">compliant</span>
                    </div>
                </div>
            </div>

            <span class="order-last self-center row-start-2 row-end-2 col-start-1 col-end-4 self-end text-left text-vanilla-400 text-xs sm:text-sm">
                © 2020, DeepSource Corp. All rights reserved.
            </span>

            <span class="row-start-2 self-center row-end-2 col-start-6 col-end-8 self-end justify-self-end text-vanilla-400 text-sm flex space-x-2 items-center">
                Backed by 
                <img class="ml-3 max-w-none" src="https://i.imgur.com/lnlRvjx.png"/>
            </span>
        </footer>
    </div>`
})
