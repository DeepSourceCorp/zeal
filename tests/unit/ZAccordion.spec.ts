import ZAccordion from '../../src/components/ZAccordion'
import ZAccordionItem from '../../src/components/ZAccordionItem'
import ZIcon from '../../src/components/ZIcon'

import { mount } from '@vue/test-utils'

const DefaultAccordion = {
  template: `<z-accordion class="text-vanilla-100">
          <z-accordion-item title="Consistency" isOpen="true">
            <div>Consistent with real life: in line with the process and logic of real life, and comply with languages and habits that the users are used to;</div>
            <div>Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc.</div>
          </z-accordion-item>
          <z-accordion-item title="Feedback">
            <div>Consistent with real life: in line with the process and logic of real life, and comply with languages and habits that the users are used to;</div>
            <div>Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc.</div>
          </z-accordion-item>
          <z-accordion-item title="Efficiency">
            <div>Consistent with real life: in line with the process and logic of real life, and comply with languages and habits that the users are used to;</div>
            <div>Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc.</div>
          </z-accordion-item>
        </z-accordion>`,
  components: {
    ZAccordion,
    ZAccordionItem
  }
}

const CustomizedAccordion = {
  template: `<div class='padded-container flex items-center w-52'>
        <z-accordion class="text-vanilla-100 w-full">
          <z-accordion-item :is-open="openAccordion">
            <template v-slot:title="{ open, toggleAccordion }">
              <div class="flex items-center cursor-pointer w-full" @click="toggleAccordion()">
                <span class="flex-1 font-medium text-slate">Resources</span>
                <z-icon icon="chevron-right" 
                    size="small" color="slate" 
                    class="transform"
                    :class="open ? 'animate-first-quarter-spin rotate-90' : 'animate-reverse-quarter-spin rotate-0'"></z-icon>
              </div>
            </template>
            <div class="flex flex-col text-sm space-y-4 py-3">
              <a v-for="link in links" :key="link" href="#" class="hover:text-juniper transition-all duration-75 ease-in-out">{{link}}</a>
            </div>
          </z-accordion-item>
        </z-accordion>
    </div>`,
  data() {
    return {
      openAccordion: false,
      links: ['Documentation', 'Discourse Form', 'Learning Center', 'Blog', 'Slack User Group']
    }
  },
  components: {
    ZAccordion,
    ZAccordionItem,
    ZIcon
  }
}

describe('Accordions', () => {
  it('renders a default accordion', () => {
    let wrapper = mount(DefaultAccordion)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders a customized accordion', () => {
    let wrapper = mount(CustomizedAccordion)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
