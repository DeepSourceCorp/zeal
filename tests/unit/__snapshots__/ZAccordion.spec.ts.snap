// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordions renders a customized accordion 1`] = `
<div class="padded-container flex items-center w-52">
  <div class="text-vanilla-100 w-full">
    <div class="z-accordion-item text-vanilla-200">
      <div class="flex items-center cursor-pointer w-full"><span class="flex-1 font-medium text-slate">Resources</span> <svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="transform z-icon w-4 h-4 z-icon--chevron-right text-slate animate-reverse-quarter-spin rotate-0">
          <polyline points="9 18 15 12 9 6"></polyline>
        </svg></div>
      <div class="overflow-hidden text-sm leading-6 duration-300 ease-in-out transition-max-height max-h-0">
        <div class="flex flex-col text-sm space-y-4 py-3"><a href="#" class="hover:text-juniper transition-all duration-75 ease-in-out">Documentation</a><a href="#" class="hover:text-juniper transition-all duration-75 ease-in-out">Discourse Form</a><a href="#" class="hover:text-juniper transition-all duration-75 ease-in-out">Learning Center</a><a href="#" class="hover:text-juniper transition-all duration-75 ease-in-out">Blog</a><a href="#" class="hover:text-juniper transition-all duration-75 ease-in-out">Slack User Group</a></div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Accordions renders a default accordion 1`] = `
<div class="text-vanilla-100">
  <div class="z-accordion-item text-vanilla-200">
    <div class="flex items-center transition-all duration-700 ease-in-out group cursor-pointer"><span class="flex-1 font-medium">Consistency</span> <svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="transform transition-all duration-DEFAULT ease-in-out group stroke-1.5 z-icon w-6 h-6 z-icon--chevron-right text-slate group-hover:text-vanilla-200">
        <polyline points="9 18 15 12 9 6"></polyline>
      </svg></div>
    <div class="overflow-hidden text-sm leading-6 duration-300 ease-in-out transition-max-height max-h-0">
      <div>Consistent with real life: in line with the process and logic of real life, and comply with languages and habits that the users are used to;</div>
      <div>Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc.</div>
    </div>
  </div>
  <div class="z-accordion-item text-vanilla-200">
    <div class="flex items-center transition-all duration-700 ease-in-out group cursor-pointer"><span class="flex-1 font-medium">Feedback</span> <svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="transform transition-all duration-DEFAULT ease-in-out group stroke-1.5 z-icon w-6 h-6 z-icon--chevron-right text-slate group-hover:text-vanilla-200">
        <polyline points="9 18 15 12 9 6"></polyline>
      </svg></div>
    <div class="overflow-hidden text-sm leading-6 duration-300 ease-in-out transition-max-height max-h-0">
      <div>Consistent with real life: in line with the process and logic of real life, and comply with languages and habits that the users are used to;</div>
      <div>Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc.</div>
    </div>
  </div>
  <div class="z-accordion-item text-vanilla-200">
    <div class="flex items-center transition-all duration-700 ease-in-out group cursor-pointer"><span class="flex-1 font-medium">Efficiency</span> <svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="transform transition-all duration-DEFAULT ease-in-out group stroke-1.5 z-icon w-6 h-6 z-icon--chevron-right text-slate group-hover:text-vanilla-200">
        <polyline points="9 18 15 12 9 6"></polyline>
      </svg></div>
    <div class="overflow-hidden text-sm leading-6 duration-300 ease-in-out transition-max-height max-h-0">
      <div>Consistent with real life: in line with the process and logic of real life, and comply with languages and habits that the users are used to;</div>
      <div>Consistent within interface: all elements should be consistent, such as: design style, icons and texts, position of elements, etc.</div>
    </div>
  </div>
</div>
`;

exports[`Accordions renders an accordion spanning full height 1`] = `
<div class="padded-container input-container">
  <div class="text-vanilla-100">
    <div class="z-accordion-item p-4 text-vanilla-200">
      <div class="flex items-center transition-all duration-700 ease-in-out group cursor-pointer"><span class="flex-1 font-medium">Span full height</span> <svg viewBox="0 0 24 24" stroke-linecap="round" stroke-linejoin="round" fill="none" class="transform transition-all duration-DEFAULT ease-in-out group stroke-1.5 z-icon w-6 h-6 z-icon--chevron-right text-slate text-vanilla-200 rotate-90 group-hover:text-vanilla-200">
          <polyline points="9 18 15 12 9 6"></polyline>
        </svg></div>
      <div class="overflow-hidden text-sm leading-6 duration-300 ease-in-out transition-max-height max-h-item">
        <p>Anim aliqua duis anim dolor ea deserunt. Consectetur aliqua labore aliquip fugiat laborum culpa adipisicing irure. Et ipsum deserunt incididunt pariatur ullamco fugiat deserunt eiusmod sit labore irure. Excepteur sint Lorem occaecat pariatur culpa fugiat eiusmod Lorem quis voluptate dolore eiusmod cupidatat est. Et quis aute eu et tempor culpa duis amet ea elit sunt. Culpa incididunt consectetur anim voluptate et.</p> <br>
        <p>Lorem irure esse reprehenderit tempor. Nisi enim commodo Lorem qui commodo officia ad ex ex. Voluptate deserunt sit et et. Irure eiusmod minim Lorem ex. Consectetur sint dolore in ullamco ad.</p> <br>
        <p>Sint voluptate in amet id. Consequat sit aute nostrud qui velit elit sunt reprehenderit elit enim non tempor aliqua occaecat. Aute quis amet nostrud dolore sunt. Commodo proident sit aliqua dolor exercitation officia anim consequat culpa exercitation dolore amet laborum. Dolor est minim do do Lorem consectetur occaecat eu officia commodo nostrud. Exercitation irure dolore deserunt laborum consectetur ad duis sunt consectetur magna elit adipisicing.</p> <br>
        <p>Velit sit voluptate tempor esse magna sint duis esse tempor. Non enim cillum incididunt irure do nisi veniam ut proident in cupidatat. Minim proident occaecat ad elit. Tempor commodo excepteur amet ad anim consectetur. Eu velit exercitation ex incididunt deserunt nisi labore consequat incididunt nostrud qui elit tempor. Pariatur aute reprehenderit dolor tempor excepteur adipisicing in. Eu culpa non est voluptate duis id.</p> <br>
        <p>Commodo incididunt aliquip duis anim id et nisi consequat. Labore ipsum do non aliqua culpa culpa laboris sint sunt laborum ad est ea. Consequat dolor nisi ut est proident veniam voluptate.</p>
      </div>
    </div>
  </div>
</div>
`;
